app/serializers/discussion_serializer.rb

```diff
@@ -36,4 +58,16 @@ def filter(keys)
     keys
   end
 
+  def scope
+    super || {}
+  end
+
+  def reader
+    @reader ||= DiscussionReader.for(user: scope[:user], discussion: object)
+  end
+
+  def has_valid_reader?
+    @has_valid_reader ||= reader.present? && reader.user.present?
```

Comment:
We check for the user because we're getting the user from scope[:user], which is nil unless we specify it explicitly.

This probably won't stay exactly like this; I'd rather have it default to 'include reader fields' rather than exclude them by default, but we can't use 'current_user' in the serializer, so we have to pass it in via the scope.